coordinator/core/marbleapi.go

```diff
@@ -107,8 +107,29 @@ func (c *Core) Activate(ctx context.Context, req *rpc.ActivationReq) (*rpc.Activ
 		SealKey:    Secret{Public: sealKey, Private: sealKey},
 	}
 
+	// Generate unique (= per marble) secrets
+	uniqueSecrets := make(map[string]Secret)
+	for k, v := range c.manifest.Secrets {
+		if v.Shared == false {
+			v.UUID = marbleUUID
```

Comment:
As every secret gets the same uuid here, I'd propose to not have this as a field in the struct, but pass it as string to generateSecrets. Then you can also remove this filtering loop and the one in SetManifest and instead skip secrets in generateSecrets based on whether uuid is empty or not.